bugsnag-android-core/src/test/java/com/bugsnag/android/RootDetectorTest.kt

```diff
@@ -96,4 +97,133 @@ class RootDetectorTest {
         val path = tmpFile.toFile().absolutePath
         assertTrue(RootDetector(rootBinaryLocations = listOf(path)).checkRootBinaries())
     }
+
+    /**
+     * Verifies that a missing file does not throw an exception
+     */
+    @Test
+    fun checkBuildPropsIOException() {
+        val tmpFile = File("/foo")
+        assertFalse(RootDetector(buildProps = tmpFile).checkBuildProps())
+    }
+
+    /**
+     * Verifies that an empty file does not trigger root detection
+     */
+    @Test
+    fun checkBuildPropsEmptyFile() {
+        val tmpFile = Files.createTempFile("empty", ".prop").toFile()
+        assertFalse(RootDetector(buildProps = tmpFile).checkBuildProps())
+    }
+
+    /**
+     * Verifies that regular build properties do not trigger root detection
+     */
+    @Test
+    fun checkBuildPropsNonRooted() {
+        val tmpFile = Files.createTempFile("regular", ".prop").toFile()
+        tmpFile.writeText(SAMPLE_BUILD_PROPS)
+        assertFalse(RootDetector(buildProps = tmpFile).checkBuildProps())
+    }
+
+    /**
+     * Verifies that the ro.secure build property triggers root detection
```

Comment:
Looks like a copy/paste error - should say root detection is not triggered.

Reply:
Updated docs to address this